### PR TITLE
#160 EA-1563 changed the total to be a number, response code.

### PR DESCRIPTION
**Pull request by rgraf**
Fixes for grouping creation

### CODE REVIEW

File: entity-web/src/main/java/eu/europeana/entity/web/controller/ConceptSchemeController.java [hunk 1]
```diff
@@ -110,6 +111,8 @@ public void setEntityService(EntityService entityService) {
 	    // not found ..
 	    // System.out.println(e);
 	    throw new InternalServerException(e);
+	} catch (ApiKeyExtractionException e) {
+	    throw new InternalServerException(HttpStatus.FORBIDDEN, e);
```
Comment (gsergiu) [inline entity-web/src/main/java/eu/europeana/entity/web/controller/ConceptSchemeController.java]: This exception is not thrown anymore in the latest version of API Commons, please remove this catch. This exception should be already converted to ApplicationAuthenticationException, that is already a HttpException

File: entity-web/src/main/java/eu/europeana/entity/web/controller/SearchController.java [hunk 2]
```diff
@@ -135,7 +137,7 @@
 	    @RequestParam(value = CommonApiConstants.QUERY_PARAM_PAGE_SIZE, required = false, defaultValue = ""
 		    + Query.DEFAULT_PAGE_SIZE) int pageSize,
 	    @RequestParam(value = CommonApiConstants.QUERY_PARAM_PROFILE, required = false) String profile,
-	    HttpServletRequest request) throws HttpException {
+	    HttpServletRequest request) throws HttpException, ApiKeyExtractionException, AuthorizationExtractionException {
```
Comment (gsergiu) [inline entity-web/src/main/java/eu/europeana/entity/web/controller/SearchController.java]: plese remove these exceptions,
 ApiKeyExtractionException and AuthorizationExtractionException are not thrown anymore in the snapshot version

File: entity-web/src/main/java/eu/europeana/entity/web/exception/InternalServerException.java [hunk 3]
```diff
@@ -23,4 +23,8 @@ public InternalServerException(Throwable th){
 		super(th.getMessage(), I18nConstants.SERVER_ERROR_UNEXPECTED, null, HttpStatus.INTERNAL_SERVER_ERROR, th);
 	}
 
+	public InternalServerException(HttpStatus httpStatus, Throwable th){
```
Comment (gsergiu) [inline entity-web/src/main/java/eu/europeana/entity/web/exception/InternalServerException.java]: please remove this constructor, this shouldn't be used

File: entity-web/src/main/java/eu/europeana/entity/web/controller/ConceptSchemeController.java (hunk 1 above)
Comment (rgraf) [reply entity-web/src/main/java/eu/europeana/entity/web/controller/ConceptSchemeController.java]: Fixed - removed local api commons project imports

File: entity-web/src/main/java/eu/europeana/entity/web/controller/SearchController.java (hunk 2 above)
Comment (rgraf) [reply entity-web/src/main/java/eu/europeana/entity/web/controller/SearchController.java]: Fixed - removed local api commons project imports

File: entity-web/src/main/java/eu/europeana/entity/web/exception/InternalServerException.java (hunk 3 above)
Comment (rgraf) [reply entity-web/src/main/java/eu/europeana/entity/web/exception/InternalServerException.java]: removed

**Pull request merged**